src/coreclr/vm/interpexec.cpp

```diff
@@ -972,6 +973,35 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                 ip += 5;
                 break;
             }
+            case INTOP_CALLVIRT:
+            {
+                returnOffset = ip[1];
+                callArgsOffset = ip[2];
+                methodSlot = ip[3];
+
+                MethodDesc *pMD = (MethodDesc*)pMethod->pDataItems[methodSlot];
+
+                OBJECTREF *pThisArg = LOCAL_VAR_ADDR(callArgsOffset, OBJECTREF);
```

Comment:
@jkotas Fixed the GC hole here, but now I'm confused how objref handling works in CoreCLR in the first place, since I can't find any uses of volatile when working with object refs. For example in the following scenario:

```
// pObj is reported to the GC via a GCFrame or some other way
void SomeMethod(OBJECTREF *pObj)
{
    // GC in COOP mode
    Use1(*pObj);
    TriggerGC();
    Use2(*pObj);
}
```

I can't seem to find what mechanism prevents the compiler from caching `*pObj` in a reg, leading to a crash at the second use.

Reply:
This particular example works via "normal" C/C++ semantics:
```
// Somewhere in the caller:
// g_globalVariable = p;

void SomeMethod(int* p)
{
    Use(*p);
    Mutate(); // Can't cache *p since it has escaped.
    Use(*p);
}

void Mutate()
{
    *g_globalVariable = <something new>;
}
```